src/Grpc.Net.ClientFactory/Internal/GrpcHttpClientFactory.cs

```diff
@@ -73,34 +74,22 @@ public TClient CreateClient(HttpClient httpClient, string name)
                 throw new ArgumentNullException(nameof(httpClient));
             }
 
-            var callInvoker = new HttpClientCallInvoker(httpClient, _loggerFactory);
-
-            var httpContext = _httpContextAccessor.HttpContext;
-            var serverCallContext = httpContext?.Features.Get<IServerCallContextFeature>().ServerCallContext;
-
-            var namedOptions = _clientOptions.Get(name);
+            var httpClientCallInvoker = new HttpClientCallInvoker(httpClient, _loggerFactory);
 
-            if (namedOptions.PropagateCancellationToken)
+            var options = _optionsMonitor.Get(name);
+            for (var i = 0; i < options.CallInvokerActions.Count; i++)
             {
-                if (serverCallContext == null)
-                {
-                    throw new InvalidOperationException("Cannot propagate the call cancellation token to the client. Cannot find the current gRPC ServerCallContext.");
-                }
-
-                callInvoker.CancellationToken = serverCallContext.CancellationToken;
+                options.CallInvokerActions[i](httpClientCallInvoker);
             }
 
-            if (namedOptions.PropagateDeadline)
+            CallInvoker resolvedCallInvoker = httpClientCallInvoker;
+            // Interceptors wrap each other and execute in reverse order
+            for (var i = options.Interceptors.Count - 1; i >= 0; i--)
             {
-                if (serverCallContext == null)
-                {
-                    throw new InvalidOperationException("Cannot propagate the call deadline to the client. Cannot find the current gRPC ServerCallContext.");
-                }
-
-                callInvoker.Deadline = serverCallContext.Deadline;
+                resolvedCallInvoker = resolvedCallInvoker.Intercept(options.Interceptors[i]);
```

Comment:
Were we okay with this? I recall we were unsure if all the functionalities here can be reproduced with just the intercept extension call. Something about modifying hosts and that it would require using the delegating handlers of HttpClient?

Reply:
You can add delegating handlers to the HttpClient alongside interceptors using the client factory.